Modules/ITS/src/TrendingTaskITSFhr.cxx

```diff
@@ -0,0 +1,327 @@
+#include <TH1.h>
+#include <map>
+#include <string>
+
+using namespace o2::quality_control;
+using namespace o2::quality_control::core;
+using namespace o2::quality_control::postprocessing;
+using namespace o2::quality_control_modules::common;
+
+void TrendingTaskITSFhr::configure(std::string name,
+                                   const boost::property_tree::ptree& config)
+{
+  mConfig = TrendingTaskConfigITS(name, config);
+}
+
+void TrendingTaskITSFhr::initialize(Trigger,
+                                    framework::ServiceRegistry& services)
+{
+  // Preparing data structure of TTree
+  mTrend = std::make_unique<TTree>(); // todo: retrieve last TTree, so we
+                                      // continue trending. maybe do it
+                                      // optionally?
+  mTrend->SetName(PostProcessingInterface::getName().c_str());
+  //mTrend->Branch("meta", &mMetaData, "runNumber/I");
+  mTrend->Branch("runNumber", &mMetaData.runNumber);
+  mTrend->Branch("ntreeentries", &ntreeentries);
+  mTrend->Branch("time", &mTime);
+
+  for (const auto& source : mConfig.dataSources) {
+    std::unique_ptr<Reductor> reductor(root_class_factory::create<Reductor>(
+      source.moduleName, source.reductorName));
+    if (source.reductorName.find("TH2Xline") != std::string::npos) {
+      TH2XlineReductor::mystat* mystc = (TH2XlineReductor::mystat*)reductor->getBranchAddress();
+      mTrend->Branch(Form("%s_mean", source.name.c_str()), &(mystc->mean));
+      mTrend->Branch(Form("%s_stddev", source.name.c_str()), &(mystc->stddev));
+      mTrend->Branch(Form("%s_entries", source.name.c_str()), &(mystc->entries));
+      mTrend->Branch(Form("%s_occupancy", source.name.c_str()), &(mystc->mean_scaled));
+    } else {
+      mTrend->Branch(source.name.c_str(), reductor->getBranchAddress(),
+                     reductor->getBranchLeafList());
+    }
+    mReductors[source.name] = std::move(reductor);
+  }
+
+  // Setting up services
+  mDatabase = &services.get<repository::DatabaseInterface>();
+}
+
+// todo: see if OptimizeBaskets() indeed helps after some time
+void TrendingTaskITSFhr::update(Trigger, framework::ServiceRegistry&)
+{
+  trendValues();
+
+  storePlots();
+  storeTrend();
+}
+
+void TrendingTaskITSFhr::finalize(Trigger, framework::ServiceRegistry&)
+{
+  storePlots();
+  storeTrend();
+}
+
+void TrendingTaskITSFhr::storeTrend()
+{
+  ILOG(Info) << "Storing the trend, entries: " << mTrend->GetEntries() << ENDM;
+
+  auto mo = std::make_shared<core::MonitorObject>(mTrend.get(), getName(),
+                                                  mConfig.detectorName);
+  mo->setIsOwner(false);
+  mDatabase->storeMO(mo);
+}
+
+void TrendingTaskITSFhr::trendValues()
+{
+  // We use current date and time. This for planned processing (not history). We
+  // still might need to use the objects
+  // timestamps in the end, but this would become ambiguous if there is more
+  // than one data source.
+  mTime = TDatime().Convert();
```

Comment:
Just to let you know, I am preparing a small modification which will add a timestamp to incoming triggers. This will eventually allow to run post-processing on any selection of timestamps, so also older data. Then you can consider taking the time from `trigger.timestamp`.

Reply:
Very good to know thanks!

Reply:
at present I added run numbers on the x axis and we will test the post processing next week with shifts. We will see how to proceed.